Python/tier2.c

```diff
@@ -2649,6 +2649,78 @@ _PyTier2_GenerateNextBB(
     return metadata->tier2_start;
 }
 
+/**
+ * @brief Helper funnction of typecontext_is_compatible. See that for why we need this.
+ * @param ctx1_node A type node belong to a type context.
+ * @param ctx2_node A type node beloning to a different type context.
+ * @return If the type nodes' parent trees are compatible.
+*/
+static bool
+typenode_is_compatible(_Py_TYPENODE_t ctx1_node, _Py_TYPENODE_t ctx2_node)
```

Comment:
This is too strict. The trees might form the same partition but have a different structure (e.g., different root).

Reply:
Hmm I'll commit a fix

Reply:
See https://github.com/pylbbv/pylbbv/issues/9#issuecomment-1574805186 for my reasoning.

Reply:
Oh yea i saw the comment, is just that, your code doesn't account for when the trees look different. E.g.,
```py
stack: [?->locals[0]]
locals: [?]

stack: [?]
locals: [?->stack[0]]
```
These two contexts should be equivalent, but won't be passed by your check

Reply:
I don't think those two contexts are equivalent. Using your example as a counterexample, for the first type context if we assign locals to be int, then stack will be int, but in the second exampls, assigning locald to be int should not affect the stack.

Reply:
But in all the type operations, in effect we should be thinking of the type trees as a convenient way of thinking of a partition of the type contexts into nodes that have the same type. We should be matching the partitions rather than the structure of the trees.